docs(upgrade): add example links to `downgradeComponent()` function docs

ajitsinghkaler opened the pull request:
There were some extra examples for downgradeComponent in the upgrade guide added the link to those examples in the downgradeComponent docs

Fixes #31584

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
The current downgradeComponent has only one example.

Issue Number: #31584


## What is the new behavior?

There were some extra examples for downgradeComponent in the upgrade guide added the link to those examples in the downgradeComponent docs


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

ajitsinghkaler commented:
@kapunahelewong can you please review this.

gkalpak commented on packages/upgrade/src/common/src/downgrade_component.ts:
```diff
@@ -40,6 +40,10 @@ import {LazyModuleRef, UpgradeAppType, controllerKey, getDowngradedModuleCount,
  *
  * {@example upgrade/static/ts/full/module.ts region="ng2-heroes-wrapper"}
+ * 
+ * For more examples and details of upgrading AngularJS Components to
```
```suggestion
 * For more details and examples on downgrading Angular components to AngularJS components please
```

gkalpak commented on packages/upgrade/src/common/src/downgrade_component.ts:
```diff
@@ -40,6 +40,10 @@ import {LazyModuleRef, UpgradeAppType, controllerKey, getDowngradedModuleCount,
  * {@example upgrade/static/ts/full/module.ts region="ng2-heroes-wrapper"}
+ * 
+ * For more examples and details of upgrading AngularJS Components to
+ * Angular Components please visit the 
```
```suggestion
 * visit the [Upgrade guide](guide/upgrade#using-angular-components-from-angularjs-code).
```

gkalpak commented on packages/upgrade/src/common/src/downgrade_component.ts:
```diff
@@ -40,6 +40,10 @@ import {LazyModuleRef, UpgradeAppType, controllerKey, getDowngradedModuleCount,
+ * 
+ * For more examples and details of upgrading AngularJS Components to
+ * Angular Components please visit the 
+ * [Upgrade guide](https://angular.io/guide/upgrade#using-angular-components-from-angularjs-code)
```
```suggestion
```

gkalpak reviewed (changes requested):
Thx, @ajitsinghkaler :+1:

I left a couple of suggestions. Can you also update the commit message to something like:

> docs(upgrade): add example links to \`downgradeComponent()\` function docs
>
> There were some extra examples for \`downgradeComponent()\` in the upgrade
> guide. Added a link to the relevant section of the guide in the
> \`downgradeComponent()\` docs.
>
> Fixes #31584

ajitsinghkaler commented:
@gkalpak I've made the changes and can you tell me more about this google3 check in ci

ajitsinghkaler committed:
docs(upgrade): add example links to `downgradeComponent()` function docs

There were some extra examples for `downgradeComponent()` in the upgrade
guide. Added a link to the relevant section of the guide in the
`downgradeComponent()` docs.

Fixes #31584

mary-poppins commented:
You can preview 8438eeb at https://pr34406-8438eeb.ngbuilds.io/.
You can preview 947e823 at https://pr34406-947e823.ngbuilds.io/.
You can preview 11ab7cc at https://pr34406-11ab7cc.ngbuilds.io/.

gkalpak commented:
👌 

> can you please tell me more about the google3 check, in ci

Don't worry about that. That is a Google-specific check, so a Google has to run it (if necessary at all).

##
@caretaker: This is a docs-only change. Maybe g3sync is not needed?
